### PR TITLE
Feature/standardize logs

### LINKED ISSUE #25: Standardize node log across all services
## 📚 Context

Each service has its own way of printing logs. For instance, here is the log for all services:

```
2023-06-07T17:32:55.132636Z  INFO dispatcher::main_loop:
2023-06-07T17:32:44.517371Z  INFO state_server_lib::utils
2023-06-07T17:32:45.964644Z  INFO advance_runner
2023-06-07T17:32:45.834129Z  INFO indexer
[2023-06-07T17:32:24Z INFO  inspect_server]
INFO graphql_server:
```

Besides the format, some have colored messages, which is problematic because some log aggregators do not parse the color information properly.

## ✔️ Solution

Here is a list of things that we should do:

- Use the same logging library (tracing).
- Disable timestamps by default. This should be a CLI/ENV option.
- Disable log colors by default. This should be a CLI/ENV option.
- The first thing in the log should be the level.

## 📈 Subtasks

Apply the modifications to all services:

- [ ] state server
- [x] dispatcher
- [x] advance runner
- [x] indexer
- [x] inspect server
- [x] graphql server
- [x] host runner

### PR DESCRIPTION
**Pull request by fmoura**
closes #25

### CODE REVIEW

**Comment by fmoura** [comment]
This is a Draft for now , but you can review as final. Only State Server should be missing

File: offchain/advance-runner/src/config.rs [hunk 1]
```diff
@@ -67,6 +73,9 @@ struct CLIConfig {
     #[command(flatten)]
     snapshot_cli_config: SnapshotCLIConfig,
 
+    #[command(flatten)]
+    pub logs_config: LogsEnvCliConfig,
```
Comment (torives) [inline offchain/advance-runner/src/config.rs:77]:
```suggestion
    pub logs_cli_config: LogsEnvCliConfig,
```
Just to use the same naming convention

File: offchain/logs/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    }
+}
+
+impl From<LogsEnvCliConfig> for LogsConfig {
+    fn from(cli_config: LogsEnvCliConfig) -> LogsConfig {
+        LogsConfig::initialize(cli_config)
+    }
+}
+
+pub fn initialize_logs(config: &LogsConfig) {
```
Comment (torives) [inline offchain/logs/src/lib.rs:47]:
```suggestion
pub fn configure(config: &LogsConfig) {
```
I'm suggesting the name change for two main reasons:

1) Losing the `logs` suffix reads better when calling the function with its fully classified name: `logs::initialize`
2) `initialize` reminds me of factory functions/methods. As we are not returning a new object from it, IMO `configure` seems a more appropriate verb.

File: offchain/host-runner/src/main.rs [hunk 3]
```diff
@@ -32,13 +32,12 @@ fn log_result<T, E: std::error::Error>(name: &str, result: Result<T, E>) {
...
 #[actix_web::main]
 async fn main() {
-    let filter = EnvFilter::builder()
-        .with_default_directive(LevelFilter::INFO.into())
-        .from_env_lossy();
-    tracing_subscriber::fmt().with_env_filter(filter).init();
+    let config: Config = CLIConfig::parse().into();
+
+    logs::initialize_logs(&config.logs_config);
+    info!("Starting Host Runner");
```
Comment (torives) [inline offchain/host-runner/src/main.rs:38]:
```suggestion
    info!(?config, "Starting Host Runner");
```
And then we can remove line 40

File: offchain/logs/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+            enable_color,
+        }
+    }
+
+    pub fn default() -> Self {
+        LogsConfig {
+            enable_timestamp: false,
+            enable_color: false,
+        }
+    }
```
[the remark this answers is not in the document]
Comment (fmoura) [reply offchain/logs/src/lib.rs:38]: I think I used something the wrong way , cause I tried to use Default. Will retry this! Thanks!!

File: offchain/advance-runner/src/config.rs (hunk 1 above)
Comment (fmoura) [reply offchain/advance-runner/src/config.rs:77]: At some projects they are like this. In others they are like those ...

Correcting this here

Comment (torives) [reply offchain/advance-runner/src/config.rs]: Yeah, it's currently messy. But #7 will hopefully solve this 😃

File: offchain/logs/Cargo.toml [hunk 5]
```diff
@@ -0,0 +1,11 @@
+[package]
+name = "logs"
+version = "1.0.0"
+edition = "2021"
+license = "Apache-2.0"
```
[the remark this answers is not in the document]
Comment (torives) [reply offchain/logs/Cargo.toml:5]: I saw you closed the conversation, but nothing changed. Did you forget to add it to the commit, by the way? 😂

**Comment by marcelstanley** [comment]
@fmoura I've made some cosmetic changes to the config files in a separate branch. Please check commit  https://github.com/cartesi/rollups-node/commit/532eb51941eb93585ff5c78d329cbb5b4f386cbe and cherry-pick it if you think it helps.

File: offchain/advance-runner/src/main.rs [hunk 6]
```diff
@@ -2,16 +2,14 @@
...
-    let filter = EnvFilter::builder()
-        .with_default_directive(LevelFilter::INFO.into())
-        .from_env_lossy();
-    tracing_subscriber::fmt().with_env_filter(filter).init();
-
     let config = AdvanceRunnerConfig::parse()?;
 
+    logs::configure(&config.logs_config);
+
+    info!(?config, "Starting Advance Runner");
```
Comment (marcelstanley) [inline offchain/advance-runner/src/main.rs:13]:
```suggestion
```

This line looks like a duplicate of the following one:

https://github.com/cartesi/rollups-node/blob/e9a9c23299a79a294d2d5ad85099a8bcdae94bfe/offchain/advance-runner/src/lib.rs#L28

File: offchain/authority-claimer/src/main.rs [hunk 7]
```diff
@@ -3,18 +3,17 @@
...
     // Settin up the logging environment.
-    let env_filter = EnvFilter::builder()
-        .with_default_directive(LevelFilter::INFO.into())
-        .from_env_lossy();
-    tracing_subscriber::fmt().with_env_filter(env_filter).init();
+    logs::configure(&config.authority_claimer_config.logs_config);
 
-    // Getting the configuration.
-    let config = Config::new().map_err(Box::new)?;
+    info!(?config, "Starting Authority Claimer");
```
Comment (marcelstanley) [inline offchain/authority-claimer/src/main.rs:16]:
```suggestion
```

This line looks like a duplicate of the following one:

https://github.com/cartesi/rollups-node/blob/e9a9c23299a79a294d2d5ad85099a8bcdae94bfe/offchain/authority-claimer/src/lib.rs#L25:

File: offchain/indexer/src/main.rs [hunk 8]
```diff
@@ -2,17 +2,17 @@
...
-    let filter = EnvFilter::builder()
-        .with_default_directive(LevelFilter::INFO.into())
-        .from_env_lossy();
-    tracing_subscriber::fmt().with_env_filter(filter).init();
+    let config: IndexerConfig = CLIConfig::parse().into();
 
-    let config = CLIConfig::parse().into();
+    logs::configure(&config.logs_config);
+
+    info!(?config, "Starting Indexer");
```
Comment (marcelstanley) [inline offchain/indexer/src/main.rs:16]:
```suggestion
```
This line looks like a duplicate of the following one:

https://github.com/cartesi/rollups-node/blob/e9a9c23299a79a294d2d5ad85099a8bcdae94bfe/offchain/indexer/src/lib.rs#L16

File: offchain/state-server/src/main.rs [hunk 9]
```diff
@@ -1,22 +1,18 @@
...
-        .with_default_directive(LevelFilter::INFO.into())
-        .from_env_lossy();
-    tracing_subscriber::fmt().with_env_filter(filter).init();
+    let config: Config = Config::initialize_from_args()?;
 
-    let config = StateServerConfig::initialize_from_args()?;
+    logs::configure(&config.logs_config);
 
-    tracing::info!(?config, "starting state server");
+    tracing::info!(?config, "Starting State Server");
```
Comment (marcelstanley) [inline offchain/state-server/src/main.rs:13]:
```suggestion
```
This line looks like a duplicate of the following one:

https://github.com/cartesi/rollups-node/blob/e9a9c23299a79a294d2d5ad85099a8bcdae94bfe/offchain/state-server/src/lib.rs#L36

File: offchain/inspect-server/src/main.rs [hunk 10]
```diff
@@ -3,16 +3,16 @@
...
-        env_logger::Env::default().default_filter_or("info"),
-    )
-    .target(env_logger::fmt::Target::Stdout)
-    .init();
-    let config = CLIConfig::parse().into();
+    let config: InspectServerConfig = CLIConfig::parse().into();
+
+    logs::configure(&config.logs_config);
+
+    info!(?config, "Starting Inpect Server");
```
Comment (marcelstanley) [inline offchain/inspect-server/src/main.rs:15]:
```suggestion
```

This line looks like a duplicate of the following one:

https://github.com/cartesi/rollups-node/blob/e9a9c23299a79a294d2d5ad85099a8bcdae94bfe/offchain/inspect-server/src/lib.rs#L18

**Review by marcelstanley — COMMENTED** [review]
@fmoura I've made some minor changes in a [separate branch](https://github.com/cartesi/rollups-node/tree/review/standardized-logs) to make it easier for me to review the PR. Feel free to cherry-pick if you think it helps.

All comments above already consider such changes.

**Review by torives — COMMENTED** [review]
Just some small tweaks and we're ready to go 😉

File: offchain/advance-runner/src/main.rs (hunk 6 above)
Comment (marcelstanley) [reply offchain/advance-runner/src/main.rs:13]: @fmoura FYI

File: offchain/authority-claimer/Cargo.toml [hunk 11]
```diff
@@ -12,9 +12,10 @@ test = false
 [dependencies]
 http-server = { path = "../http-server" }
 rollups-events = { path = "../rollups-events" }
+log = { path = "../log" }
```
Comment (marcelstanley) [inline offchain/authority-claimer/Cargo.toml:15]: Just keeping them in alphabetic order:
```suggestion
log = { path = "../log" }
rollups-events = { path = "../rollups-events" }
```

File: offchain/authority-claimer/src/main.rs (hunk 7 above)
Comment (marcelstanley) [reply offchain/authority-claimer/src/main.rs:16]: @fmoura FYI

File: offchain/dispatcher/Cargo.toml [hunk 12]
```diff
@@ -13,6 +13,7 @@ contracts = { path = "../contracts" }
 http-server = { path = "../http-server" }
 rollups-events = { path = "../rollups-events" }
 types = { path = "../types" }
+log = {path = "../log"}
```
Comment (marcelstanley) [inline offchain/dispatcher/Cargo.toml:16]: Just keeping them in alphabetic order:

```suggestion
log = { path = "../log" }
rollups-events = { path = "../rollups-events" }
types = { path = "../types" }
```

File: offchain/graphql-server/Cargo.toml [hunk 13]
```diff
@@ -15,6 +15,7 @@ path = "src/schema/generate_schema.rs"
 [dependencies]
 http-health-check = { path = "../http-health-check" }
 rollups-data = { path = "../data" }
+log = { path = "../log" }
```
Comment (marcelstanley) [inline offchain/graphql-server/Cargo.toml:18]: Just keeping them in alphabetic order:
```suggestion
log = { path = "../log" }
rollups-data = { path = "../data" }
```

File: offchain/indexer/Cargo.toml [hunk 14]
```diff
@@ -13,6 +13,7 @@ test = false
 http-health-check = { path = "../http-health-check" }
 rollups-data = { path = "../data" }
 rollups-events = { path = "../rollups-events" }
+log = {path = "../log"}
```
Comment (marcelstanley) [inline offchain/indexer/Cargo.toml:16]: Also keeping them in alphabetic order:
```suggestion
log = { path = "../log" }
rollups-data = { path = "../data" }
rollups-events = { path = "../rollups-events" }
```

File: offchain/indexer/src/main.rs (hunk 8 above)
Comment (marcelstanley) [reply offchain/indexer/src/main.rs:16]: @fmoura FYI

File: offchain/log/Cargo.toml [hunk 15]
```diff
@@ -0,0 +1,11 @@
+[package]
+name = "log"
+version = "1.0.0"
+edition = "2021"
+license = "Apache-2.0"
```
Comment (marcelstanley) [inline offchain/log/Cargo.toml:5]:
```suggestion
edition.workspace = true
license.workspace = true
version.workspace = true
```

File: offchain/state-server/src/config.rs [hunk 16]
```diff
@@ -0,0 +1,40 @@
+use clap::Parser;
+use eth_state_server_lib::config::{
+    Result, StateServerConfig, StateServerEnvCLIConfig,
+};
+use log::{LogConfig, LogEnvCliConfig};
+
+#[derive(Debug, Clone, Parser)]
+#[command(name = "sate_server_config")]
```
Comment (marcelstanley) [inline offchain/state-server/src/config.rs:8]:
```suggestion
#[command(name = "state_server_config")]
```

File: offchain/state-server/Cargo.toml [hunk 17]
```diff
@@ -9,8 +9,9 @@ name = "cartesi-rollups-state-server"
 path = "src/main.rs"
 
 [dependencies]
+log = { path = "../log" }
```
Comment (marcelstanley) [inline offchain/state-server/Cargo.toml:12]: Keeping them in alphabetic order:

```suggestion
clap = { workspace = true, features = ["derive", "env"] }
```

File: offchain/state-server/Cargo.toml [hunk 18]
```diff
@@ -9,8 +9,9 @@ name = "cartesi-rollups-state-server"
...
 types = { path = "../types" }
-
+clap = { workspace = true, features = ["derive", "env"] }
```
Comment (marcelstanley) [inline offchain/state-server/Cargo.toml:14]: Keeping them in alphabetic order:
```suggestion
log = { path = "../log" }
```

**Review by marcelstanley — CHANGES_REQUESTED** [review]

File: offchain/inspect-server/src/main.rs (hunk 10 above)
Comment (marcelstanley) [reply offchain/inspect-server/src/main.rs]: @fmoura FYI

File: CHANGELOG.md [hunk 19]
```diff
@@ -9,6 +9,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Added
 
+- Standardize log libraries and log configurations
 - Added authority claimer service to support reader mode
 - Added support to `POST` *inspect state* requests
```
[the remark this answers is not in the document]
Comment (marcelstanley) [reply CHANGELOG.md]: Done at 9e1b6f8e7bc477c334f4e109cf3bb7a1642bc44d

File: offchain/dispatcher/Cargo.toml (hunk 12 above)
[the remark this answers is not in the document]
Comment (marcelstanley) [reply offchain/dispatcher/Cargo.toml]: Done at 9e1b6f8e7bc477c334f4e109cf3bb7a1642bc44d

File: offchain/advance-runner/Cargo.toml [hunk 20]
```diff
@@ -12,6 +12,7 @@ path = "src/main.rs"
 grpc-interfaces = { path = "../grpc-interfaces" }
 http-health-check = { path = "../http-health-check" }
 rollups-events = { path = "../rollups-events" }
+log = {path = "../log"}
```
[the remark this answers is not in the document]
Comment (marcelstanley) [reply offchain/advance-runner/Cargo.toml]: Done at 9e1b6f8e7bc477c334f4e109cf3bb7a1642bc44d

File: offchain/indexer/Cargo.toml (hunk 14 above)
[the remark this answers is not in the document]
Comment (marcelstanley) [reply offchain/indexer/Cargo.toml]: Done at 9e1b6f8e7bc477c334f4e109cf3bb7a1642bc44d

File: offchain/advance-runner/tests/fixtures/mod.rs [hunk 21]
```diff
@@ -90,13 +91,19 @@ impl AdvanceRunnerFixture {
...
+        };
+
         let config = AdvanceRunnerConfig {
             server_manager_config,
             broker_config,
             dapp_metadata,
             snapshot_config,
             backoff_max_elapsed_duration,
             healthcheck_port: 0,
+            log_config,
```
[the remark this answers is not in the document]
Comment (marcelstanley) [reply offchain/advance-runner/tests/fixtures/mod.rs]: Done at 9e1b6f8e7bc477c334f4e109cf3bb7a1642bc44d

File: offchain/indexer/tests/integration.rs [hunk 22]
```diff
@@ -442,11 +443,16 @@ async fn spawn_indexer(
...
+        enable_timestamp: false,
+        enable_color: false,
+    };
     let indexer_config = indexer::IndexerConfig {
         repository_config,
         dapp_metadata,
         broker_config,
         healthcheck_port: 0,
+        log_config,
     };
```
[the remark this answers is not in the document]
Comment (marcelstanley) [reply offchain/indexer/tests/integration.rs]: Done at 9e1b6f8e7bc477c334f4e109cf3bb7a1642bc44d

File: offchain/authority-claimer/Cargo.toml (hunk 11 above)
Comment (marcelstanley) [reply offchain/authority-claimer/Cargo.toml]: Done at 9e1b6f8e7bc477c334f4e109cf3bb7a1642bc44d

File: offchain/dispatcher/Cargo.toml (hunk 12 above)
Comment (marcelstanley) [reply offchain/dispatcher/Cargo.toml]: Done at 9e1b6f8e7bc477c334f4e109cf3bb7a1642bc44d

File: offchain/indexer/Cargo.toml (hunk 14 above)
Comment (marcelstanley) [reply offchain/indexer/Cargo.toml]: Done at 9e1b6f8e7bc477c334f4e109cf3bb7a1642bc44d

File: offchain/log/Cargo.toml (hunk 15 above)
Comment (marcelstanley) [reply offchain/log/Cargo.toml]: Done at 9e1b6f8e7bc477c334f4e109cf3bb7a1642bc44d

File: offchain/state-server/src/config.rs (hunk 16 above)
Comment (marcelstanley) [reply offchain/state-server/src/config.rs]: Done at 9e1b6f8e7bc477c334f4e109cf3bb7a1642bc44d

File: offchain/state-server/Cargo.toml (hunk 17 above)
Comment (marcelstanley) [reply offchain/state-server/Cargo.toml:12]: Done at 9e1b6f8e7bc477c334f4e109cf3bb7a1642bc44d

File: offchain/graphql-server/Cargo.toml (hunk 13 above)
Comment (marcelstanley) [reply offchain/graphql-server/Cargo.toml]: Done at 9e1b6f8e7bc477c334f4e109cf3bb7a1642bc44d

File: offchain/inspect-server/Cargo.toml [hunk 23]
```diff
@@ -10,13 +10,14 @@ path = "src/main.rs"
 
 [dependencies]
 http-health-check = { path = "../http-health-check" }
+log = { path = "../log" }
+
+
```
[the remark this answers is not in the document]
Comment (marcelstanley) [reply offchain/inspect-server/Cargo.toml]: Done at b84e916d871f07526378131f1df7e8d8e8586759

File: offchain/state-server/src/main.rs (hunk 9 above)
Comment (marcelstanley) [reply offchain/state-server/src/main.rs:13]: @fmoura this one still applies. I'll push a fixup

**Comment by marcelstanley** [comment]
Perhaps @torives would like to take a last quick look 👀

**Review by torives — APPROVED** [review]
LGTM 🚀

**Pull request merged**